backend/api.py

```diff
@@ -0,0 +1,152 @@
+""" RESTful API for http://tuixue.online/global/"""
+
+from enum import Enum
+from typing import Optional, List
+from datetime import datetime, timedelta
+
+from fastapi import FastAPI, Body, Query, Response, status
+from fastapi.middleware.cors import CORSMiddleware
+from pydantic import BaseModel
+
+import global_var as G
+import tuixue_mongodb as DB
+from notifier import Notifier
+from tuixue_typing import VisaType, EmbassyCode
+
+EMBASSY_LST = G.USEmbassy.get_embassy_lst()
+app = FastAPI()
+app.add_middleware(
+    CORSMiddleware,
+    allow_origin_regex=r'http[s]://localhost:?[\d]*',
```

Comment:
If I recall it correctly, this is for developing the React frontend, when you running the React server in you local machine with a backend api server at the same time. The origin will be `http://localhost:{React-port}`.

When users fetch data from their browser this can be unnecessary, we can change it anytime when developing frontend

Reply:
okay